jsk_robot_common/jsk_robot_utils/euslisp/robot-teleop-joy.l

```diff
@@ -191,6 +253,76 @@
         *head-neck-y-min-angle* (+ (send *robot* :head-neck-y :min-angle) 1.0))
   ;; spin rate
   (ros::rate 50)
+
   ;; subscribe joy topic
   (ros::subscribe "/joy" sensor_msgs::Joy #'teleop-joy-cb)
   )
+
+
+(defun robot-teleop-joy-setup(&key
+                              (robot "pr2") (joy-type "ps3") (grasp-loop-enable nil)
+                              (reset-loop-enable nil) (look-at-loop-enable nil) (move-head-enable nil))
+  (let ((reset-pose :reset-manip-pose))
+    (cond
+     ((string= robot "hrp2")
+      (load "package://hrpsys_ros_bridge_tutorials/euslisp/hrp2jsknts-interface.l")
+      (defmethod hrp2jsknts-interface
+        (:start-grasp (arm &key (robot *robot*) (time 3000) (wait nil) &allow-other-keys)
+                      (send robot :hand arm :index-avoid-grasp-pose)
+                      (send self :hand-angle-vector (send robot :hand-angle-vector) time)
+                      (if wait (send self :hand-wait-interpolation)))
+        (:stop-grasp (arm &key (robot *robot*) (time 3000) (wait nil) &allow-other-keys)
+                     (send robot :hand arm :hook-pose)
+                     (send self :hand-angle-vector (send robot :hand-angle-vector) time)
+                     (if wait (send self :hand-wait-interpolation))))
+      (hrp2jsknts-init)
+      (setq *robot* *hrp2jsknts*))
+     ((string= robot "pr2")
+      (load "package://pr2eus/pr2-interface.l")
+      (pr2-init)
+      (setq *robot* *pr2*))
+     ((string= robot "aero")
+      (load "package://aeroeus/aero-interface.l")
+      (aero-init)
+      (setq *robot* *aero*)
+      (setq reset-pose :reset-pose)))
+
+    (when *debug-teleop-joy*
+      (unless (boundp '*irtviewer*)
+        (make-irtviewer)
+        (objects (list *robot*))))
+
+    ;; setup teleop-joy
+    (teleop-joy-setup)
+    (setq *joy-type* joy-type)
+    ;; options
+    (setq *grasp-loop-enable* grasp-loop-enable
+          *reset-loop-enable* reset-loop-enable
+          *look-at-loop-enable* look-at-loop-enable
+          *move-head-enable* move-head-enable)
+
+    reset-pose))
+
+(defun setup ()
+  (let ((node-ns (ros::get-name))
+        (reset-pose :reset-manip-pose))
+    (setq *max-pos* (ros::get-param (concatenate string node-ns "/max_pos") 3.0))
```

Comment:
You can delete `defvar`s in top of this file in order to avoid duplicated definitions.